openmc/source.py

```diff
@@ -270,7 +270,7 @@ def particle(self):
 
     @particle.setter
     def particle(self, particle):
-        cv.check_value('source particle', particle, ['neutron', 'photon'])
+        cv.check_value('source particle', particle, ['neutron', 'photon', 'random_ray'])
```

Comment:
I really do not like the idea of random_ray being another type of particle. Since there is a long-term ambition to have RR generate importance maps, we'd need both multigroup neutron and photon transport for random ray. If a user wishes to set both neutron and photon sources separately in a random ray simulation, it wouldn't jive with this way of distinguishing track types. There would be a potential desire to distinguish between neutron random rays and photon random rays. Or any other particle if we added more!

Reply:
The enum is just being used to let the user specify which source should be used to sample ray starting locations vs. which sources are actually corresponding to fixed source neutrons and/or photon distributions. Originally I was just enforcing that a single source would need to be present in random ray mode, and it would use that source for sampling the rays, but after working ahead in implementing the fixed source solver this wasn't a good solution. In the C++ side of things, rays would not be generated with their enums set to random ray, so they should be able to do either neutron or photon transport already. With multigroup transport, I'm not sure there really is any distinction between neutrons/photons, other than whatever you did when generating the cross sections.

As for coupled neutron + photon simultaneously, that would be something to think about down the line potentially. There would be nothing from stopping us (as you said in another comment) from having the rays integrate both spectrums at the same time, amortizing the ray tracing/geometry costs. Would just need an extra update function that executes when preparing the sources for the next iteration that relates neutron flux into the photon source term.

Reply:
On this topic -- I did notice that the vacuum BC was checking for the random ray particle type, which was unintended. It should instead be checking for the solver mode, as `RandomRay` objects aren't actually using the type field currently.

Reply:
OK, yes, after digesting it a bit, this all makes sense.